.github/workflows/delete-workflow.yaml

```diff
@@ -0,0 +1,44 @@
+name: Delete old workflow runs
+on:
+  workflow_dispatch:
+    inputs:
+      days:
+        description: "Number of days."
+        required: true
+        default: 30
+      minimum_runs:
+        description: "The minimum runs to keep for each workflow."
+        required: true
+        default: 6
+      delete_workflow_pattern:
+        description: "The name or filename of the workflow. if not set then it will target all workflows."
+        required: false
+      delete_workflow_by_state_pattern:
+        description: "Remove workflow by state: active, deleted, disabled_fork, disabled_inactivity, disabled_manually"
+        required: true
+        default: "All"
+        type: choice
+        options:
+          - "All"
+          - active
+          - deleted
+          - disabled_inactivity
+          - disabled_manually
+      dry_run:
+        description: "Only log actions, do not perform any delete operations."
+        required: false
+
+jobs:
+  del_runs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Delete workflow runs
+        uses: Mattraks/delete-workflow-runs@v2
```

Comment:
In apache/pulsar, it's only possible to use whitelisted GitHub Actions because of security reasons. That is why we have copies of the actions added in https://github.com/apache/pulsar-test-infra after a careful review. Another option would be to review a specific revision of the action and send that information to Apache Infra administrators for whitelisting. The pulsar-test-infra solution is what we have been using. For example https://github.com/apache/pulsar-test-infra/tree/master/paths-filter includes a copy of https://github.com/dorny/paths-filter .

Reply:
Sure, I'll add the action to apache/pulsar-test-infra.